### PR TITLE
[run] #230 Add `--sanitizer` flag

### LINKED ISSUE #230: Compile generators/validators separately from submissions
Not sure how, but it would be nice to set compilation flags/commands for submissions only

### PR DESCRIPTION
**Pull request by mzuenni**
Fixes #230.

### CODE REVIEW

**Review by mpsijm — APPROVED** [review]
Looking good! I also tried running `bt run` on some problem, with and without the `--sanitizer` flag, and it properly recompiles the submissions between the runs :slightly_smiling_face: I don't know how to trigger sanitizer warnings, but that will probably work fine, since this also worked with the old `--cpp_flags=-fsanatize=undefined`. :smile:

File: bin/program.py [hunk 1]
```diff
@@ -29,14 +29,20 @@
     run: '{run}'
 '''
 
+SANITIZER_FLAGS = '''
+cpp:
+    compile: -fsanitize=undefined
```
Comment (RagnarGrootKoerkamp) [inline bin/program.py:34]: wanna add `,address` also here?

Comment (mzuenni) [reply bin/program.py:34]: i get an error if i add this...
```
Exited with code -6:
==618896==ERROR: AddressSanitizer failed to allocate 0xdfff0001000 (15392894357504) bytes at address 2008fff7000 (errno: 12)
==618896==ReserveShadowMemoryRange failed while trying to map 0xdfff0001000 bytes. Perhaps you're using ulimit -v
```

Comment (RagnarGrootKoerkamp) [reply bin/program.py:34]: ah yes, `ulimit` / memory limit should be disabled for that. But that probably makes sense to do right?

Comment (mzuenni) [reply bin/program.py:34]: The issue seems to be that we set a memory limit. If we want to have `-fsanitize=address` we have to skip the memory limit. I can add that if you want?

**Pull request merged**